### PR TITLE
[CHANGING][MISC] Scale the default joint armature with the inertia each joint drives.

### PR DESCRIPTION
**Pull request by duburcqa**
## Description

The default joint armature is now a fraction (0.1) of the joint-space sub-tree inertia each degree of freedom drives at the initial configuration, instead of a flat 0.1 kg.m^2. It applies at build to every revolute or prismatic joint moving a link on its own whose armature the asset leaves at zero; authored values are kept. The parsers no longer inject the default, heterogeneous entities resolve it per variant and environment, and MuJoCo compatibility is unchanged.

The default constraint solver tolerance in single precision drops to 1e-7 in every mode, the largest value at which light bodies resting beside a robot still settle once its joints carry less armature. Double precision keeps 1e-9, and MuJoCo's 1e-8 under compatibility.

The MJCF parser honors the joint-level `actuatorfrcrange`, which it dropped before, so models stating their torque limits that way ran unclamped; every stated bound (actuator force range, motor control range through its gear, joint range) applies at once.

The finite-difference gradient checks use a fourth-order Richardson extrapolation, compare against the shared `tol` fixture, and read outputs whose gradients clear that tolerance in single precision: velocity for the per-step force checks, faster spins in the all-groups scene. Checks whose signal is structurally below the fp32 tolerance (forces into an active limit, pose under per-step forces) run in double precision only.

The g1_fall benchmark runs at the default tolerance and keeps a uniform 0.1 kg.m^2 armature on the G1 joints, with a TODO recording that neither the robot's actuator limits nor its published rotor inertia make the scenario integrable at its timestep.

## Related Issue

Addresses Genesis-Embodied-AI/Genesis#3257: the reported scene is not a grasp (the sphere starts 5 cm inside the fingers and MuJoCo ejects it too), but a valid pinch that MuJoCo compatibility holds was dropped under default options because of the flat default armature.

## Motivation and Context

A flat 0.1 kg.m^2 is a sensible tenth of an arm joint's inertia but four orders of magnitude above a finger's, so a dexterous hand under default options cannot react to contact within a step and drops what it holds. Authored armatures across the MuJoCo Menagerie sit between 0.01 and 1 times the driven inertia; the relative default lands every model in that band. Joints the flat armature used to freeze now move, so scenes driving light joints hard see more active constraints and run slower (dex_hand loses about 13%), none of it from the tolerance default.

## How Has This Been / Can This Be Tested?

`test_default_armature` checks the fraction, the kept authored values and the per-variant resolution. `test_general_actuator` states joint-level force ranges on a saturating PD joint and a geared motor, asserts the parsed ranges and gains, and compares the clamped actuator forces against MuJoCo step by step. The gradient tests fail against the fixture tolerance at the previous steps and pass at the new ones.

## Checklist:

- [x] I read the **CONTRIBUTING** document.
- [x] I followed the `Submitting Code Changes` section of **CONTRIBUTING** document.
- [x] I tagged the title correctly (including BUG FIX/FEATURE/MISC/BREAKING)
- [x] I updated the [documentation](https://github.com/Genesis-Embodied-AI/genesis-doc) accordingly or no change is needed.
- [x] I tested my changes and added instructions on how to test it for reviewers.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: genesis/engine/solvers/rigid/rigid_solver.py [hunk 1]
```diff
@@ -399,6 +400,39 @@ def build(self):
...
+        if dofs_idx:
+            is_default = torch.atleast_2d(self.get_dofs_armature(dofs_idx) <= 0.0).all(dim=0)
+            if is_default.any():
+                dofs_idx = torch.tensor(dofs_idx, dtype=gs.tc_int, device=gs.device)[is_default]
+                dofs_ratio = torch.tensor(np.stack(dofs_ratio, axis=1), dtype=gs.tc_float, device=gs.device)
+                mass_mat_diag = torch.atleast_2d(self.get_mass_mat().diagonal(dim1=-2, dim2=-1)[..., dofs_idx])
+                dofs_armature = dofs_ratio[:, is_default] * mass_mat_diag
+                if not self._options.batch_dofs_info or self.n_envs == 0:
+                    dofs_armature = dofs_armature[0]
+                self.set_dofs_armature(dofs_armature, dofs_idx)
```
[the remark this answers is not in the document]
Comment (duburcqa) [reply genesis/engine/solvers/rigid/rigid_solver.py]: fixed

**Comment by duburcqa** [comment]
@codex review

File: genesis/engine/solvers/rigid/rigid_solver.py [hunk 2]
```diff
@@ -399,6 +402,51 @@ def build(self):
...
+            is_default = (np.atleast_2d(dofs_armature[dofs_idx].T) <= 0.0).all(axis=0)
+            if is_default.any():
+                dofs_idx = np.array(dofs_idx)[is_default]
+                dofs_ratio = np.stack(dofs_ratio, axis=1)[:, is_default]
+                mass_mat_diag = np.atleast_2d(tensor_to_array(self.get_mass_mat().diagonal(dim1=-2, dim2=-1)))
+                default_armature = dofs_ratio * mass_mat_diag[:, dofs_idx]
+                dofs_armature[dofs_idx] = default_armature.T if self._options.batch_dofs_info else default_armature[0]
+                self.dyn_info.dofs.armature.from_numpy(dofs_armature)
+                dofs_invweight = qd_to_numpy(self.dyn_info.dofs.invweight, transpose=False, copy=True)
+                dofs_invweight[dofs_idx] = -1.0
```
[the remark this answers is not in the document]
Comment (duburcqa) [reply genesis/engine/solvers/rigid/rigid_solver.py]: fixed

**Comment by duburcqa** [comment]
@codex review

File: genesis/engine/solvers/rigid/rigid_solver.py [hunk 3]
```diff
@@ -399,6 +402,53 @@ def build(self):
...
+                dofs_idx = np.array(dofs_idx)[is_default]
+                dofs_ratio = np.stack(dofs_ratio, axis=1)[:, is_default]
+                mass_mat_diag = np.atleast_2d(tensor_to_array(self.get_mass_mat().diagonal(dim1=-2, dim2=-1)))
+                default_armature = dofs_ratio * mass_mat_diag[:, dofs_idx]
+                dofs_armature[dofs_idx] = default_armature.T if self._options.batch_dofs_info else default_armature[0]
+                self.dyn_info.dofs.armature.from_numpy(dofs_armature)
+                defaulted = (entity for entity, is_dof_default in zip(dofs_entity, is_default) if is_dof_default)
+                entities = dict.fromkeys(defaulted)
+                dofs_invweight = qd_to_numpy(self.dyn_info.dofs.invweight, transpose=False, copy=True)
+                dofs_invweight[[i_d for entity in entities for i_d in range(entity.dof_start, entity.dof_end)]] = -1.0
```
[the remark this answers is not in the document]
Comment (duburcqa) [reply genesis/engine/solvers/rigid/rigid_solver.py]: fixed
